r/R/table.R

```diff
@@ -149,6 +149,77 @@ Table$create <- function(..., schema = NULL) {
 #' @export
 names.Table <- function(x) x$ColumnNames()
 
+#' @export
+rbind.Table <- function(...) {
+  tables <- list(...)
+
+  # assert they have same schema
+  schema <- tables[[1]]$schema
+  unequal_schema_idx <- which.min(lapply(tables, function(x) x$schema == schema))
+  if (unequal_schema_idx != 1) {
+    abort(c(
+      sprintf("Schema at index %i does not match the first schema", unequal_schema_idx),
+      i = paste0("Schema 1:\n", schema$ToString()),
+      i = paste0(sprintf("Schema %d:\n", unequal_schema_idx),
+                 tables[[unequal_schema_idx]]$schema$ToString())
+    ))
+  }
+
+  # create chunked array for each column
+  columns <- map(seq_len(tables[[1]]$num_columns), function(i) {
+    do.call(c, map(tables, ~ .[[i]]))
+  })
+
+  Table$create(!!!set_names(columns, names(schema)), schema = schema)
+}
+
+#' @export
+cbind.Table <- function(...) {
```

Comment:
This is basically the same as cbind.RecordBatch, right? Any way we can consolidate those two?

Reply:
Yes, except in `cbind.Table`, chunked arrays and tables are allowed. `cbind_check_length` was an effort to consolidate.

I just simplified them a lot though so the duplication isn't as bad.

Reply:
> Yes, except in `cbind.Table`, chunked arrays and tables are allowed.

Two thoughts here, feel free to ignore or defer to a followup:

1. Given how you've simplified, I wonder if that's a problem. If you passed through chunked arrays to RecordBatch$create, it should error, and hopefully with an informative message. In fact if I read the latest version correctly, cbind.RecordBatch isn't asserting that it doesn't get chunked arrays, so you're already relying on that behavior. 
2. Maybe @paleolimbot's as_* S3 generics PR would help? cbind.RecordBatch would `as_arrow_array` its input columns, and cbind.Table would `as_chunked_array` its inputs.

Reply:
I agree with (2). If I understand right, that would mean that `cbind` would play nicely with types that convert into extension arrays, which would be great.